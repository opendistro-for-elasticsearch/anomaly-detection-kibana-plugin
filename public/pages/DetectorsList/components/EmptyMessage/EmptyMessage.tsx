/*
 * Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License").
 * You may not use this file except in compliance with the License.
 * A copy of the License is located at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * or in the "license" file accompanying this file. This file is distributed
 * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 * express or implied. See the License for the specific language governing
 * permissions and limitations under the License.
 */

import { EuiButton, EuiEmptyPrompt, EuiText } from '@elastic/eui';
import React from 'react';
import { APP_PATH, PLUGIN_NAME } from '../../../../utils/constants';

const filterText =
  'There are no detectors matching your applied filters. Reset your filters to view all detectors.';
const emptyText =
  'Anomaly detectors take an input of information and discover patterns of anomalies. Create an anomaly detector to get started.';

interface EmptyDetectorProps {
  isFilterApplied: boolean;
  onResetFilters: () => void;
}

export const EmptyDetectorMessage = (props: EmptyDetectorProps) => (
  <EuiEmptyPrompt
    style={{ maxWidth: '45em' }}
    body={
      <EuiText>
        <p>{props.isFilterApplied ? filterText : emptyText}</p>
      </EuiText>
    }
    actions={
      props.isFilterApplied ? (
        <EuiButton
          fill
          onClick={props.onResetFilters}
          data-test-subj="reset_list_filters"
        >
          Reset filters
        </EuiButton>
      ) : (
        <EuiButton
          fill
          href={`${PLUGIN_NAME}#${APP_PATH.CREATE_DETECTOR}`}
          data-test-subj="add_detector"
        >
          Create detector
        </EuiButton>
      )
    }
  />
);
