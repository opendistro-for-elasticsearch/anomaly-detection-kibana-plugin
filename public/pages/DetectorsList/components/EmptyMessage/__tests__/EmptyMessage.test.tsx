/*
 * Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License").
 * You may not use this file except in compliance with the License.
 * A copy of the License is located at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * or in the "license" file accompanying this file. This file is distributed
 * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 * express or implied. See the License for the specific language governing
 * permissions and limitations under the License.
 */

import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { EmptyDetectorMessage } from '../EmptyMessage';

describe('<EmptyDetectorMessage /> spec', () => {
  describe('Empty results', () => {
    test('renders component with empty message', () => {
      const { container, getByText } = render(
        <EmptyDetectorMessage
          isFilterApplied={false}
          onResetFilters={jest.fn()}
        />
      );
      expect(container.firstChild).toMatchSnapshot();
      getByText('Create detector');
    });
  });
  describe('Filters results message', () => {
    test('renders component no result for filters message', () => {
      const { container } = render(
        <EmptyDetectorMessage
          isFilterApplied={true}
          onResetFilters={jest.fn()}
        />
      );
      expect(container.firstChild).toMatchSnapshot();
    });
    test('resets filters when click on reset filters', () => {
      const handleResetFilters = jest.fn();
      const { getByTestId } = render(
        <EmptyDetectorMessage
          isFilterApplied={true}
          onResetFilters={handleResetFilters}
        />
      );
      fireEvent.click(getByTestId('resetListFilters'));
      expect(handleResetFilters).toHaveBeenCalled();
      expect(handleResetFilters).toHaveBeenCalledTimes(1);
    });
  });
});
