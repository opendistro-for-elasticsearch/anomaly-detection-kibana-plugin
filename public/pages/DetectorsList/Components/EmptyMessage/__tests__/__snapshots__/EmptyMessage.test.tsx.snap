// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EmptyDetectorMessage /> spec Empty results renders component with empty message 1`] = `
<div
  class="euiEmptyPrompt"
  style="max-width: 45em;"
>
  <span
    class="euiTextColor euiTextColor--subdued"
  >
    <div
      class="euiText euiText--medium"
    >
      <div
        class="euiText euiText--medium"
      >
        <p>
          Anomaly detectors take an input of information and discover patterns of anomalies. Create an anomaly detector to get started.
        </p>
      </div>
    </div>
  </span>
  <div
    class="euiSpacer euiSpacer--l"
  />
  <div
    class="euiSpacer euiSpacer--s"
  />
  <a
    class="euiButton euiButton--primary euiButton--fill"
    data-test-subj="add_detector"
    href="opendistro-anomaly-detection#/create-ad/"
    rel="noreferrer"
  >
    <span
      class="euiButton__content"
    >
      <span
        class="euiButton__text"
      >
        Create detector
      </span>
    </span>
  </a>
</div>
`;

exports[`<EmptyDetectorMessage /> spec Filters results message renders component no result for filters message 1`] = `
<div
  class="euiEmptyPrompt"
  style="max-width: 45em;"
>
  <span
    class="euiTextColor euiTextColor--subdued"
  >
    <div
      class="euiText euiText--medium"
    >
      <div
        class="euiText euiText--medium"
      >
        <p>
          There are no detectors matching your applied filters. Reset your filters to view all detectors.
        </p>
      </div>
    </div>
  </span>
  <div
    class="euiSpacer euiSpacer--l"
  />
  <div
    class="euiSpacer euiSpacer--s"
  />
  <button
    class="euiButton euiButton--primary euiButton--fill"
    data-test-subj="reset_list_filters"
    type="button"
  >
    <span
      class="euiButton__content"
    >
      <span
        class="euiButton__text"
      >
        Reset filters
      </span>
    </span>
  </button>
</div>
`;
